### PR TITLE
8 patch

### PR DESCRIPTION
**Pull request by sweepas**
Section from 8 to 12

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+  test("Should return 200 and and vote count shoud be incrumented by 1", () => {
+    const body = { inc_votes: 1 };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(body)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.votes).toBe(101);
+        expect(body).toHaveProperty("title");
+        expect(body).toHaveProperty("article_id");
```
Comment (pippaamy) [inline __tests__/app.test.js]: we can put that this is article 1

File: __tests__/app.test.js [hunk 2]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+      .then(({ body }) => {
+        expect(body.votes).toBe(50);
+        expect(body.title).not.toBe("changed title");
+      });
+  });
+  test("Should ignore irrelevant inputs", () => {
+    const body = {
+      inc_votes: "not votes",
+    };
+    return request(app)
```
Comment (pippaamy) [inline __tests__/app.test.js:279]: this could potentially be a 200 and just ignore anything not important

File: __tests__/app.test.js [hunk 3]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+      title: "changed title",
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(body)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.votes).toBe(50);
+        expect(body.title).not.toBe("changed title");
+      });
```
Comment (pippaamy) [inline __tests__/app.test.js:273]: we need a couple of error tests for if article id doesnt exist/ invalid id

File: __tests__/app.test.js [hunk 4]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+    });
+  });
+});
+describe("DELETE /api/comments/:comment_id", () => {
+  test("should return 204 and no content", () => {
+    return request(app).delete("/api/comments/1").expect(204);
+  });
+  test("should return 404 not found if provided with not existing ID", () => {
+    return request(app)
+      .delete("/api/comments/999")
```
Comment (pippaamy) [inline __tests__/app.test.js:312]: great tests for delete

File: __tests__/app.test.js [hunk 5]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+        body.users.forEach((user) => {
+          expect(user).toHaveProperty("username");
+          expect(user).toHaveProperty("name");
+          expect(user).toHaveProperty("avatar_url");
+        });
+      });
+  });
+  test("should get 404 bad request", () => {
+    return request(app)
+      .get("/api/not-users")
```
Comment (pippaamy) [inline __tests__/app.test.js:343]: good testing, you may have a similar test to this before

File: __tests__/app.test.js [hunk 6]
```diff
@@ -218,3 +219,155 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+        expect(body.articles.length).toBe(12);
+        body.articles.forEach((article) => {
+          expect(article).not.toHaveProperty("body");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).toHaveProperty("comment_count");
+          expect(article).toHaveProperty("author");
```
Comment (pippaamy) [inline __tests__/app.test.js:366]: how about if topic=paper which exists but returns an empty array

**Review by pippaamy — COMMENTED** [review]
great job, some changes to add and pop in another pr :)

**Review by Jodiech — APPROVED** [review]
Nice work on amending your tests for this ticket, please go ahead and merge! 🚀

**Pull request merged**